dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AuthenticatedResourceInterface.java

```diff
@@ -210,4 +216,27 @@ default void optionalUserCheckEntry(Optional<User> user, Entry entry) {
             }
         }
     }
+
+    default void mutateBasedOnUserAgent(Entry entry, ManipulateEntry m, ContainerRequestContext containerContext) {
+        try {
+            final List<String> strings = containerContext.getHeaders().get("User-Agent");
```

Comment:
What happens if user-agent is not set? Does this return an empty list or null?

Reply:
Looks like null, resolving with getOrDefault